src/main/java/address/browser/BrowserManager.java

```diff
@@ -58,32 +61,36 @@
     };
 
     public BrowserManager(ObservableList<ReadOnlyViewablePerson> filteredPersons) {
-        initializeBrowser();
         this.selectedPersonUsername = new SimpleStringProperty();
         this.filteredPersons = filteredPersons;
     }
 
     /**
-     * Initialize the application to use jxBrowser.
+     * Initialize the browser managed by the browser manager.
      * This must be called in a non-ui thread.
      */
-    public static void initializeJxBrowserEnvironment(){
-        if (Environment.isMac()) {
-            BrowserCore.initialize();
+    public static void initBrowser(){
+        if (type == BrowserType.FULL_FEATURE_BROWSER) {
+            if (Environment.isMac()) {
+                BrowserCore.initialize();
+            }
+            logger.debug("Suppressing browser logs");
+            LoggerProvider.setLevel(Level.SEVERE);
         }
```

Comment:
Do you mean to replace it with embeddedBrowser.init() here and implement the init method in respective concrete class of embeddedBrowser? The embeddedbrowser instance will only be created at the start() method. In which, the UI-thread will be started. And this block of codes need to run on a non-UI thread.